core/src/main/scala/pureconfig/DerivedReaders.scala

```diff
@@ -20,9 +20,26 @@ import shapeless.labelled._
 trait AllowMissingKey { self: ConfigReader[_] => }
 
 /**
+ * Trait extending [[DerivedReaders1]] that contains `ConfigReader` instances for `AnyVal`.
+ *
+ * This trait exists to give priority to the `AnyVal` derivation over the generic product derivation.
+ */
+trait DerivedReaders extends DerivedReaders1 {
```

Comment:
You can also remove the ambiguity by requiring a `F <:!< AnyVal` evidence in `deriveProductInstance`. Have you considered it? I am not sure of which way I prefer.